ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ClassicBlockProcessor.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.mainnet;
+
+import org.hyperledger.besu.ethereum.chain.Blockchain;
+import org.hyperledger.besu.ethereum.core.Address;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+import org.hyperledger.besu.ethereum.core.MutableAccount;
+import org.hyperledger.besu.ethereum.core.MutableWorldState;
+import org.hyperledger.besu.ethereum.core.ProcessableBlockHeader;
+import org.hyperledger.besu.ethereum.core.Transaction;
+import org.hyperledger.besu.ethereum.core.TransactionReceipt;
+import org.hyperledger.besu.ethereum.core.Wei;
+import org.hyperledger.besu.ethereum.core.WorldUpdater;
+import org.hyperledger.besu.ethereum.mainnet.MainnetBlockProcessor.TransactionReceiptFactory;
+import org.hyperledger.besu.ethereum.vm.BlockHashLookup;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class ClassicBlockProcessor implements BlockProcessor {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private static final int MAX_GENERATION = 6;
+  private static final long ERA_LENGTH = 5_000_000L;
+
+  private final TransactionProcessor transactionProcessor;
+
+  private final TransactionReceiptFactory transactionReceiptFactory;
+
+  private final Wei blockReward;
+
+  private final boolean skipZeroBlockRewards;
+
+  private final MiningBeneficiaryCalculator miningBeneficiaryCalculator;
+
+  public ClassicBlockProcessor(
+      final TransactionProcessor transactionProcessor,
+      final TransactionReceiptFactory transactionReceiptFactory,
+      final Wei blockReward,
+      final MiningBeneficiaryCalculator miningBeneficiaryCalculator,
+      final boolean skipZeroBlockRewards) {
+    this.transactionProcessor = transactionProcessor;
+    this.transactionReceiptFactory = transactionReceiptFactory;
+    this.blockReward = blockReward;
+    this.miningBeneficiaryCalculator = miningBeneficiaryCalculator;
+    this.skipZeroBlockRewards = skipZeroBlockRewards;
+  }
+
+  @Override
+  public Result processBlock(
```

Comment:
This should be refactored into an AbstractBlockProcessor.  Between this one and MainnetBlockProcessor this method is identical, aside from fully qualified references to `Result`.  That class should be in the parent too.  `rewardCoinbase` should be an abstract method and Mainnet and Classic should do their calculation in the subclasses.

Reply:
I agree.... This has been updated.